docs/explanation/parts.rst

```diff
@@ -0,0 +1,263 @@
+specifies a location where the source code or other information is to be
+*pulled* from, and may be a repository on a remote server, a directory on
+the build host, or some other location.
+
+Additional keys are used to fine-tune the specification so that a precise
+description of the source location can be given, and also to specify the type
+of source to be processed.
+
+Where the type of the source information cannot be automatically determined,
+the ``source-type`` key is used to explicitly specify the source format.
+This influences the way in which the source code or data is processed.
+A list of supported formats can be found in the `craft_parts/sources/sources.py` file. Examples include "bzr", "git", "hg", "mercurial", "subversion", "svn", "tar", "zip", "deb", "rpm", "7z", "local".
+
+If the source type represents a file, the ``source-checksum`` key can be used
+to provide a checksum value to be compared against the checksum of the
+downloaded file.
+
+Parts with source types that describe repositories can also use additional
+keys to accurately specify where source code is found. The ``source-branch``,
+``source-commit`` and ``source-tag`` keys allow sources to be obtained from
+a specific branch, commit or tag.
+
+Since some repositories can contain large amounts of data, the
+``source-depth`` key can be used to specify the number of commits in a
+repository's history that should be fetched instead of the complete history.
+For repositories that use submodules, the ``source-submodules`` key can be
+used to selectively fetch only those submodules that are needed.
+
+The ``source-subdir`` key specifies the subdirectory in the unpacked sources
+where builds will occur. **Note:** This key restricts the build to the
+subdirectory specified, preventing access to files in the parent directory
+and elsewhere in the file system directory structure.
+
+Dependencies
+~~~~~~~~~~~~
+
+The dependencies of a part are described using the ``build-snaps`` and
+``build-packages`` definitions. These specify lists of snaps and system
+packages to be installed before the part is built.
+
+System packages are referred to by the names that they are identified by on
+the host system, and they are installed using the host's native package
+manager, such as :command:`apt` or :command:`dnf`.
+
+Build process
+~~~~~~~~~~~~~
+
+Every part needs to define how it will be built. Since all parts are built
+using *plugins*, each part definition will use the ``plugin`` key.
+
+The available plugins are provided as modules in the `craft_parts/plugins`
+directory -- see :py:mod:`craft_parts.plugins`. These aim to simplify the
+process of building source code written in a variety of programming languages
+using appropriate build systems, libraries and frameworks. If a plugin is not
+available for a particular combination of these attributes, a custom plugin
+can be created or a basic plugin can be used to manually specify the build
```

Comment:
The application using craft-parts can create new plugins, but the text seems to suggest that these can be created by the final user and that's not supported. Previous versions of Snapcraft had user-defined "local plugins", but craft-parts doesn't support custom plugins created by the user.

Reply:
OK. That's very useful to know. Thanks.